src/components/Toast/useToast.tsx

```diff
@@ -0,0 +1,18 @@
+import { useCallback } from "react";
+import { useToastContext } from "./ToastContext";
+import { ToastNoticeProps } from "./ToastNotice";
+
+export interface UseToastHook {
+  showToast: (props: ToastNoticeProps) => void;
+}
+export function useToast(): UseToastHook {
+  const { setNotice } = useToastContext();
+
+  const showToast = useCallback((props) => {
+    setNotice({
+      ...props,
+    });
```

Comment:
Suggestion: Really minor but maybe `setNotice(props)`? Not that the copy will be expensive, but just that maintainers might wonder "...why do we need a copy here?", and if there isn't an explicit reason, they might get distracted by it w/o needing to be.

(Alternatively, if the copy is actually required, an in-source comment about why would be great.)